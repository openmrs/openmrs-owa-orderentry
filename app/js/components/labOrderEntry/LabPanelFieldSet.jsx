import React from 'react';
import PropTypes from 'prop-types';
import { panelData } from './labData';
import '../../../css/grid.scss';

class LabPanelFieldSet extends React.Component {
  render() {
    const { selectedPanelIds, handleTestSelection } = this.props;
    return (
      <fieldset className="fieldset">
        <legend>Panels</legend>
        {
          panelData.map(panel => (
            <button
              id="panel-button"
              className={(selectedPanelIds.includes(panel.id)) ? 'active lab-tests-btn' : 'lab-tests-btn'}
              type="button"
              key={`${panel.id}`}
              onClick={() => handleTestSelection(panel, 'panel')}
            >
              {panel.name}
            </button>
          ))
        }
      </fieldset>
    );
  }
}

LabPanelFieldSet.defaultProps = {
  selectedPanelIds: [],
};

LabPanelFieldSet.propTypes = {
  handleTestSelection: PropTypes.func.isRequired,
  selectedPanelIds: PropTypes.array,
};

export default LabPanelFieldSet;
