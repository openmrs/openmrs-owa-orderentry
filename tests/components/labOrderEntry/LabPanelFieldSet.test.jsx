import React from 'react';
import LabPanelFieldSet from '../../../app/js/components/labOrderEntry/LabPanelFieldSet';

let props;
let mountedComponent;
props = {
  selectTests: jest.fn(),
  selectPanel: jest.fn()
};

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<LabPanelFieldSet { ...props } />);
  }
  return mountedComponent;
};

describe('Component: LabPanelFieldSet', () => {
  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should mount initially', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should support click for each button rendered', () => {
    const component = getComponent();
    const instance = component.instance();
    sinon.spy(instance, 'setPanel');
    const panelButton = component.find('#panel-button').at(0); // click the second button
    panelButton.simulate('click', {});
    expect(instance.setPanel.calledOnce).toEqual(true);
  });
});
