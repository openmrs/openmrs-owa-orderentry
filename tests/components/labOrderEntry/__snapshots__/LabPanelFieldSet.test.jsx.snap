// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LabPanelFieldSet should mount initially 1`] = `
<fieldset
  className="fieldset"
>
  <legend>
    Panels
  </legend>
  <div
    className="panel-box"
  >
    <button
      className="lab-tests-btn tooltip"
      id="panel-button"
      key="asampleduuid1234"
      onClick={[Function]}
      type="button"
    >
      sample
      <ToolTip
        toolTipBody={
          Array [
            "sample",
          ]
        }
        toolTipHeader="Tests included in this panel:"
      />
    </button>
  </div>
</fieldset>
`;
